File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02905/p02905_num7_parsed.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#define MOD 998244353

using namespace std;

// Function to calculate LCM of two numbers
int lcm(int a, int b) {
    return (a / __gcd(a, b)) * b;
}

int main() {
    int N;
    cin >> N;
    
    vector<int> A(N);
    for (int i = 0; i < N; i++) {
        cin >> A[i];
    }
    
    long long sum = 0;
    for (int i = 0; i < N - 1; i++) {
        for (int j = i + 1; j < N; j++) {
            sum += lcm(A[i], A[j]);
            sum %= MOD;
        }
    }
    
    cout << sum << endl;
    
    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02905/p02905_num7_parsed.py
from math import gcd

N = int(input())
A = list(map(int, input().split()))

MOD = 998244353

# calculate the least common multiple of two numbers
def lcm(a, b):
    return (a * b) // gcd(a, b)

# calculate the sum of least common multiples
def sum_lcm(A):
    n = len(A)
    result = 0
    for i in range(n-1):
        for j in range(i+1, n):
            result += lcm(A[i], A[j])
            result %= MOD
    return result

answer = sum_lcm(A)
print(answer)